### PR TITLE
 Add `focus` method of <Mention />

### LINKED ISSUE #6135: Mention提供focus方法
### What problem does this feature solve?
Mention的用户交互上尽可能的接近输入框，应当提供设置focus的功能
我们的一个场景是：用户点击某个div后，会将隐藏的Mention组件显示出来，如果用户点击Mention组件之外的区域，Mention又将会被重新隐藏，这就需要我们能够监听onBlur，问题是，当Mention显示出来时，如果用户不主动点击Mention区域内时，Mention无法处于focus状态，后续点击区域外的时候也无法响应onBlur

### What does the proposed API look like?
function focus() like Input

### PR DESCRIPTION
**Pull request by RaoHai**
+ close #6135

### CODE REVIEW

File: components/mention/index.tsx [hunk 1]
```diff
@@ -101,6 +103,16 @@ export default class Mention extends React.Component<MentionProps, MentionState>
       this.props.onBlur(ev);
     }
   }
+  focus = () => {
+    this.mentionEle._editor.focus();
+  }
+  mentionRef = ele => {
+    const { ref } = this.props;
```
Comment (benjycui) [inline components/mention/index.tsx]: I don't understand, it seems that we [cannot read ref from props](https://github.com/facebook/react/blob/1c449cea348bc18add78c5892af332e6febd38b5/src/isomorphic/classic/element/ReactElement.js#L208).

Comment (afc163) [inline components/mention/index.tsx]: @RaoHai

**Comment by benjycui** [comment]
I will handle this issue.

**Comment by benjycui** [comment]
Trace: https://github.com/ant-design/ant-design/pull/6643

**Pull request closed without merging**